aws/rust-runtime/aws-inlineable/src/s3_express.rs

```diff
@@ -56,15 +62,109 @@ pub(crate) mod auth {
     impl Sign for S3ExpressSigner {
         fn sign_http_request(
             &self,
-            _request: &mut HttpRequest,
-            _identity: &Identity,
-            _auth_scheme_endpoint_config: AuthSchemeEndpointConfig<'_>,
-            _runtime_components: &RuntimeComponents,
-            _config_bag: &ConfigBag,
+            request: &mut HttpRequest,
+            identity: &Identity,
+            auth_scheme_endpoint_config: AuthSchemeEndpointConfig<'_>,
+            runtime_components: &RuntimeComponents,
+            config_bag: &ConfigBag,
         ) -> Result<(), BoxError> {
-            todo!()
+            let operation_config =
+                SigV4Signer::extract_operation_config(auth_scheme_endpoint_config, config_bag)?;
+            let mut settings = SigV4Signer::signing_settings(&operation_config);
+
+            let express_credentials = identity.data::<Credentials>().ok_or(
+                "wrong identity type for SigV4. Expected AWS credentials but got `{identity:?}",
+            )?;
+
+            add_token_to_request(express_credentials, request, &mut settings)?;
+
+            SigV4Signer.sign_http_request(
+                request,
+                identity,
+                settings,
+                &operation_config,
+                runtime_components,
+                config_bag,
+            )
         }
     }
+
+    fn add_token_to_request(
+        express_credentials: &Credentials,
+        request: &mut HttpRequest,
+        settings: &mut SigningSettings,
+    ) -> Result<(), BoxError> {
+        match settings.signature_location {
+            SignatureLocation::Headers => {
+                let security_token_header = Cow::Borrowed("x-amz-security-token");
```

Comment:
The STS session token must be excluded from signing, but doesn't it also need to be excluded from the request entirely? Shouldn't we be removing it from the headers instead of excluding it from signing here?

Reply:
Addressed 830ef02.